external-resolver/src/main/scala/org/genivi/sota/resolver/filters/FilterDirectives.scala

```diff
@@ -27,54 +27,68 @@ import slick.jdbc.JdbcBackend.Database
+    parameter('regex.as[String Refined Regex].?) { re =>
+      val query = re.fold(FilterRepository.list)(re => FilterRepository.searchByRegex(re))
+      complete(db.run(query))
+    }
 
-    pathPrefix("filters") {
-      (get & pathEnd) {
-        parameter('regex.as[Refined[String, Regex]].?) { re =>
-          val query = re.fold(FilterRepository.list)(re => FilterRepository.searchByRegex(re))
-          complete(db.run(query))
-        }
-      } ~
-      (get & refined[Filter.ValidName](Slash ~ Segment) & path("package")) { filter =>
-        completeOrRecoverWith(db.run(PackageFilterRepository.listPackagesForFilter(filter))) {
-          Errors.onMissingFilter
-        }
-      } ~
-      (post & entity(as[Filter]) & pathEnd) { filter =>
-        complete(db.run(FilterRepository.add(filter)))
-      } ~
-      (put & refined[Filter.ValidName](Slash ~ Segment ~ PathEnd)
-           & entity(as[Filter.ExpressionWrapper])
-           & pathEnd)
-      { (fname, expr) =>
-        complete(db.run(FilterRepository.update(Filter(fname, expr.expression))))
-      } ~
-      (delete & refined[Filter.ValidName](Slash ~ Segment ~ PathEnd) & pathEnd)
-      { fname =>
-        complete(db.run(FilterRepository.deleteFilterAndPackageFilters(fname)))
-      }
+  def getPackages(fname: String Refined Filter.ValidName) =
+    completeOrRecoverWith(db.run(PackageFilterRepository.listPackagesForFilter(fname))) {
+      Errors.onMissingFilter
+    }
+
+  def createFilter =
+    entity(as[Filter]) { filter =>
+      complete(db.run(FilterRepository.add(filter)))
+    }
 
+  def updateFilter(fname: String Refined Filter.ValidName) =
+    entity(as[Filter.ExpressionWrapper]) { expr =>
+      complete(db.run(FilterRepository.update(Filter(fname, expr.expression))))
     }
 
+  def deleteFilter(fname: String Refined Filter.ValidName) =
+    complete(db.run(FilterRepository.deleteFilterAndPackageFilters(fname)))
+
   /**
    * API route for validating filters.
    * @return      Route object containing routes for verifying that a filter is valid
    */
-  def validateRoute: Route = {
-    pathPrefix("validate") {
-      path("filter") ((post & entity(as[Filter])) (_ => complete("OK")))
+  def validateFilter =
+    entity(as[Filter]) { filter =>
+      complete("OK")
     }
-  }
 
+  /**
+   * API route for filters.
+   * @return      Route object containing routes for getting, creating,
+   *              editing, deleting, and validating filters
+   * @throws      Errors.MissingFilterException if the filter doesn't exist
+   */
   def route: Route =
-    handleExceptions( ExceptionHandler( Errors.onMissingFilter orElse Errors.onMissingPackage ) ) {
-      filterRoute ~ validateRoute
+    handleExceptions(ExceptionHandler(Errors.onMissingFilter orElse Errors.onMissingPackage)) {
+      pathPrefix("filters") {
+        (get & pathEnd) {
+          searchFilter
+        } ~
+        (get & refined[Filter.ValidName](Slash ~ Segment) & path("package")) { fname =>
```

Comment:
The same here for this refined, so we can use it in ota plus. (`~ PathEnd`) can go out of the refined call into a normal `pathEnd`.